File: backend/src/AssetManager.Domain/Entities/Asset.cs
namespace AssetManager.Domain.Entities;

public class Asset
{
    public int Id { get; set; }
    public int UserId { get; set; }
    public required string Name { get; set; }
    public AssetType Type { get; set; }
    public decimal Quantity { get; set; }
    public decimal CostBasis { get; set; }
    public decimal CurrentPrice { get; set; }
    public string Currency { get; set; } = "USD";
    public string? Notes { get; set; }
    public bool IsDeleted { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

    public User User { get; set; } = null!;
    public CryptoDetails? CryptoDetails { get; set; }
    public StockDetails? StockDetails { get; set; }
    public RealEstateDetails? RealEstateDetails { get; set; }
}

File: frontend/src/types/asset.ts
export type AssetType = "Crypto" | "Stock" | "RealEstate";
export type PropertyType = "House" | "Apartment" | "Commercial" | "Land";

export interface CryptoDetails {
  symbol: string;
  network?: string | null;
  walletAddress?: string | null;
  exchange?: string | null;
  staking: boolean;
}

export interface StockDetails {
  symbol: string;
  exchange?: string | null;
  sector?: string | null;
  dividendYield: number;
}

export interface RealEstateDetails {
  propertyType: PropertyType;
  address: string;
  city: string;
  state: string;
  zipCode?: string | null;
  country?: string | null;
  purchasePrice: number;
  currentValue: number;
  mortgageBalance: number;
  monthlyRent: number;
  monthlyExpenses: number;
  squareFeet?: number | null;
  bedrooms?: number | null;
  bathrooms?: number | null;
  yearBuilt?: number | null;
}

export interface Asset {
  id: number;
  name: string;
  type: AssetType;
  quantity: number;
  costBasis: number;
  currentPrice: number;
  currency: string;
  totalValue: number;
  totalCost: number;
  gainLoss: number;
  gainLossPercent: number;
  notes?: string | null;
  createdAt: string;
  updatedAt: string;
  cryptoDetails?: CryptoDetails | null;
  stockDetails?: StockDetails | null;
  realEstateDetails?: RealEstateDetails | null;
}

export interface AssetListResponse {
  items: Asset[];
  total: number;
  page: number;
  pageSize: number;
  hasMore: boolean;
}
